packages/api/src/external/carequality/document/create-outbound-document-query-req.ts

```diff
@@ -21,34 +22,78 @@ export function createOutboundDocumentQueryRequests({
   cxId: string;
   cqLinks: CQLink[];
 }): OutboundDocumentQueryReq[] {
-  const now = dayjs().toISOString();
+  const now = dayjs();
   const user = getSystemUserName(initiator.orgName);
 
-  return cqLinks.map(externalGateway => {
-    return {
-      id: requestId,
-      cxId: cxId,
-      timestamp: now,
-      samlAttributes: {
-        subjectId: user,
-        subjectRole: {
-          code: SUBJECT_ROLE_CODE,
-          display: SUBJECT_ROLE_DISPLAY,
+  return cqLinks.flatMap(externalGateway => {
+    if (doesGatewayNeedDateRanges(externalGateway.url)) {
```

Comment:
There's a lot of logic here, can we move this to a function that we call in the `.flatMap()`, please?

Reply:
Also, both "sides" of the `if` build the request pretty much the same, just the first one adds `documentCreationDate`. Maybe we create them first, and then check if the requests need to be further broken down and updated?